go/zeusmaster/zeusmaster_test.go

```diff
@@ -0,0 +1,265 @@
+package zeusmaster_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/burke/zeus/go/config"
+	slog "github.com/burke/zeus/go/shinylog"
+	"github.com/burke/zeus/go/unixsocket"
+	"github.com/burke/zeus/go/zeusmaster"
+)
+
+var testFiles = map[string]string{
+	"zeus.json": `
+{
+  "command": "ruby -r./custom_plan -eZeus.go",
+  "plan": {
+    "boot": {
+      "data": {
+        "data_srv": {}
+      },
+      "code": {
+        "code_srv": {}
+      },
+      "cmd_srv": []
+    }
+  }
+}
+`,
+	"custom_plan.rb": `
+$LOAD_PATH.unshift(File.readlink('./lib'))
+require 'zeus'
+
+class CustomPlan < Zeus::Plan
+  def self.command(name, &block)
+    define_method(name) do
+      redirect_log(name)
+      begin
+        self.instance_eval(&block)
+      rescue => e
+        STDERR.puts "#{name} terminated with exception: #{e.message}"
+        STDERR.puts e.backtrace.map {|line| " #{line}"}
+        raise
+      end
+    end
+  end
+
+  command :boot do
+    redirect_log('boot')
+    require_relative 'srv'
+  end
+
+  command :data do
+    redirect_log('data')
+    require_relative 'data'
+  end
+
+  command :code do
+    redirect_log('code')
+    require_relative 'code'
+  end
+
+  command :cmd_srv do
+    redirect_log('cmd_srv')
+    serve('cmd.sock')
+  end
+
+  command :data_srv do
+    redirect_log('data_srv')
+    serve('data.sock')
+  end
+
+  command :code_srv do
+    redirect_log('code_srv')
+    serve('code.sock')
+  end
+
+  def redirect_log(cmd)
+    log_file = File.open("zeus_test_#{cmd}.log", 'a')
+    log_file.sync = true
+    STDOUT.reopen(log_file)
+    STDERR.reopen(log_file)
+    STDOUT.sync = STDERR.sync = true
+  end
+end
+
+Zeus.plan = CustomPlan.new
+`,
+	"data.rb": `
+require 'yaml'
+$response = YAML::load_file('data.yaml')['response']
+`,
+	"data.yaml": `
+response: YAML the Camel is a Mammal with Enamel
+`,
+	"other-data.yaml": `
+response: Hi
+`,
+	"code.rb": `
+$response = "Hello, world!"
+`,
+	"other-code.rb": `
+$response = "there!"
+`,
+	"srv.rb": `
+$response = "pong"
+
+def serve(sock_path)
+  sock = Socket.new(Socket::AF_UNIX, Socket::SOCK_DGRAM, 0)
```

Comment:
This is a great first cut! If you make this a connected STREAM socket, you could detect reconnection & could use three deadlines below (expectation that the connection closes, that another one comes up, and that something gets sent).